backends/consul_client.go

```diff
@@ -0,0 +1,94 @@
+package backends
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+)
+
+// Client provides a wrapper around the consulkv client
+type ConsulClient struct {
+	client *api.KV
+}
+
+// NewConsulClient returns a new client to Consul for the given address
+func NewConsulClient(config *Config) (*ConsulClient, error) {
```

Comment:
What's the circular import issue here? IMO the `backends` package is meant to be a [factory package](http://en.wikipedia.org/wiki/Factory_method_pattern) with each client implemented in its own `backends/<backend>` package, so we really should be following the same pattern:

```go
package consul

func New(config *Config) (*ConsulClient, error) {
...
}
```